cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -170,40 +178,52 @@ impl YaccParser {
                 let mut avoid_insert = HashSet::new();
                 i = self.parse_ws(j, false)?;
                 let num_newlines = self.newlines.len();
+                if self.ast.avoid_insert.is_none() {
+                    self.ast.avoid_insert = Some(HashSet::new());
+                }
                 while j < self.src.len() && self.newlines.len() == num_newlines {
                     let (j, n) = self.parse_token(i)?;
                     self.ast.tokens.insert(n.clone());
+                    if avoid_insert.contains(&n)
```

Comment:
I reckon we should remove the `avoid_insert` variable and just stuff things in `self.ast.avoid_insert` directly, which will simplify this `if`.

Reply:
Good idea!